File: 브론즈 2 (5문제)/1964_오각형, 오각형, 오각형 ....py
# 참고 링크 : https://moon-gd.tistory.com/29
dot = 0

# 단계 입력받기
N = int(input())

dot = (N + 1) * N // 2 + (N + 1) ** 2

print(dot % 45678)

File: 브론즈 2 (5문제)/1964_오각형, 오각형, 오각형.c
// 참고 링크 :  https://moon-gd.tistory.com/29
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>

int main() {
	long long dot = 0, N;

	scanf("%lld", &N);

	dot = (N + 1) * N / 2 + (N + 1) * (N + 1);
	dot %= 45678;
	printf("%lld", dot);

	return 0;
}